src/Microsoft.ComponentDetection.Detectors/pip/Contracts/PipInstallationMetadata.cs

```diff
@@ -0,0 +1,79 @@
+namespace Microsoft.ComponentDetection.Detectors.Pip;
+
+using Newtonsoft.Json;
+
+/// <summary>
+/// Metadata for a pip component being installed. See https://packaging.python.org/en/latest/specifications/core-metadata/.
+/// Some fields are not collected here because they are not needed for dependency graph construction.
+/// </summary>
+public class PipInstallationMetadata
```

Comment:
```suggestion
public sealed record PipInstallationMetadata
```

Reply:
Same for all the other new contracts.

Sealed brings significant performance improvements[^1]

[^1]: https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-6/#peanut-butter

Reply:
All new contracts converted to sealed records 👍